src/main/java/net/snowflake/client/core/JsonSqlInput.java

```diff
@@ -55,90 +59,93 @@ public JsonNode getInput() {
             mapSFExceptionToSQLException(() -> converters.getNumberConverter().getByte(value)));
   }
 
   @Override
   public short readShort() throws SQLException {
-    return withNextValue(
-        (value, jsonNode, fieldMetadata) -> {
-          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
-          return mapSFExceptionToSQLException(
-              () -> converters.getNumberConverter().getShort(value, columnType));
-        });
+    return withNextValue((value, fieldMetadata) -> convertShort(value, fieldMetadata));
+  }
+
+  private Short convertShort(Object value, FieldMetadata fieldMetadata) throws SQLException {
+    int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+    return mapSFExceptionToSQLException(
+        () -> converters.getNumberConverter().getShort(value, columnType));
   }
 
   @Override
   public int readInt() throws SQLException {
-    return withNextValue(
-        (value, jsonNode, fieldMetadata) -> {
-          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
-          return mapSFExceptionToSQLException(
-              () -> converters.getNumberConverter().getInt(value, columnType));
-        });
+    return withNextValue((value, fieldMetadata) -> convertInt(value, fieldMetadata));
+  }
+
+  private Integer convertInt(Object value, FieldMetadata fieldMetadata) throws SQLException {
+    int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+    return mapSFExceptionToSQLException(
+        () -> converters.getNumberConverter().getInt(value, columnType));
   }
 
   @Override
   public long readLong() throws SQLException {
-    return withNextValue(
-        (value, jsonNode, fieldMetadata) -> {
-          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
-          return mapSFExceptionToSQLException(
-              () -> converters.getNumberConverter().getLong(value, columnType));
-        });
+    return withNextValue((value, fieldMetadata) -> convertLong(value, fieldMetadata));
+  }
+
+  private Long convertLong(Object value, FieldMetadata fieldMetadata) throws SQLException {
+    int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
+    return mapSFExceptionToSQLException(
+        () -> converters.getNumberConverter().getLong(value, columnType));
   }
 
   @Override
   public float readFloat() throws SQLException {
-    return withNextValue(
-        (value, jsonNode, fieldMetadata) -> {
-          int columnType = ColumnTypeHelper.getColumnType(fieldMetadata.getType(), session);
-          return mapSFExceptionToSQLException(
-              () -> converters.getNumberConverter().getFloat(value, columnType));
-        });
+    return withNextValue((value, fieldMetadata) -> convertFloat(value, fieldMetadata));
+  }
+
+  private Float convertFloat(Object value, FieldMetadata fieldMetadata) throws SQLException {
```

Comment:
All these functions look very similar to the ones from ArrowSqlInput. Maybe we can pull them to parent class?